dango/dex/src/query.rs

```diff
@@ -142,10 +142,34 @@ fn query_orders_by_pair(
 
 #[inline]
 fn query_orders_by_user(
-    _ctx: ImmutableCtx,
-    _user: Addr,
-    _start_after: Option<OrderId>,
-    _limit: Option<u32>,
+    ctx: ImmutableCtx,
+    user: Addr,
+    start_after: Option<OrderId>,
+    limit: Option<u32>,
 ) -> StdResult<BTreeMap<OrderId, OrdersByUserResponse>> {
-    todo!();
+    let start = start_after.map(|order_id| {
+        let (((base_denom, quote_denom), direction, price, _), _) =
+            ORDERS.idx.order_id.load(ctx.storage, order_id).unwrap();
```

Comment:
We can't unwrap here, otherwise an attack can query an non-existing order id, then this would panic and crash the node. We need to avoid all sorts of panics in `RustVm` contracts.

Reply:
Good point